### PR TITLE
[OSE-292] SubmitApplication now returns an Operation

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
A credential application needs to be reviewed (by an automated or manual process). This asynchronous nature is better suited to be a long running operation.

# Description
This is part of the changes needed to support manual and automated issuance of credentials after an application is submitted.
Some of the changes include moving files into their own packages. This was done so break dependency cycles.

# How Has This Been Tested?
This is a refactor. Tests still pass (with some slight modifications)

# Checklist

Before submitting this PR, please make sure:

- [x] I have read the CONTRIBUTING document.
- [x] My code is consistent with the rest of the project 
- [x] I have tagged the relevant reviewers and/or interested parties
- [x] I have updated the READMEs and other documentation of affected packages

## References
SIP5

### CODE REVIEW

File: pkg/server/router/operation.go [hunk 1]
```diff
@@ -188,3 +201,22 @@ func (o OperationRouter) CancelOperation(ctx context.Context, w http.ResponseWri
...
+	Result OperationResult `json:"result,omitempty"`
+}
+
+type OperationResult struct {
+	// Populated when there was an error with the operation.
+	Error string `json:"error,omitempty"`
+
+	// Populated iff Error == "". The type should be specified in the calling APIs documentation.
+	Response any `json:"response,omitempty"`
+}
```
Comment (andresuribe87) [inline pkg/server/router/operation.go:222]: This was moved from `pkg/server/router/presentation.go`

File: pkg/service/operation/credential/response.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package credential
```
Comment (nitro-neal) [inline pkg/service/operation/credential/response.go:1]: this file is basically the same as the presentation/response file right? (except the parentResource)

Is there anyway we can combine, or do you want it to be separate so they can implement different status if they want to

File: pkg/server/router/operation.go [hunk 3]
```diff
@@ -188,3 +201,22 @@ func (o OperationRouter) CancelOperation(ctx context.Context, w http.ResponseWri
...
 }
+
+type Operation struct {
+	// The name of the resource related to this operation. E.g. "presentations/submissions/<uuid>"
+	ID string `json:"id"`
+
+	// Whether this operation has finished.
+	Done bool `json:"done"`
+
+	// Populated Done == true.
```
Comment (nitro-neal) [inline pkg/server/router/operation.go:212]: // Populated if Done == true.

File: pkg/server/router/manifest.go [hunk 4]
```diff
@@ -313,20 +314,14 @@ func (mr ManifestRouter) SubmitApplication(ctx context.Context, w http.ResponseW
...
 	}
 
-	resp := SubmitApplicationResponse{
-		Response:    submitApplicationResponse.Response,
-		Credentials: submitApplicationResponse.Credentials,
-		ResponseJWT: submitApplicationResponse.ResponseJWT,
-	}
-
-	return framework.Respond(ctx, w, resp, http.StatusCreated)
+	return framework.Respond(ctx, w, routerModel(*op), http.StatusCreated)
```
Comment (nitro-neal) [inline pkg/server/router/manifest.go:324]: eventually we can create a routerModel everywhere

**Review by nitro-neal — APPROVED** [review]
Looks great!

It seems like steelthread integration test still works with minimal changes.

In the future we should make another steelthread test that tests the asynchronous flow when it is ready

File: pkg/server/router/manifest.go (hunk 4 above)
Comment (andresuribe87) [reply pkg/server/router/manifest.go:324]: For `Operation` objects, yes!

File: pkg/server/router/operation.go (hunk 3 above)
Comment (andresuribe87) [reply pkg/server/router/operation.go]: Done.

File: pkg/service/operation/credential/response.go (hunk 2 above)
Comment (andresuribe87) [reply pkg/service/operation/credential/response.go:1]: Status is already slightly different: rejected vs denied, fulfilled vs approved. 

I think there is a chance of unifying, but I'm not comfortable yet deciding what the correct generalization would look like.

**Review by andresuribe87 — COMMENTED** [review]
> In the future we should make another steelthread test that tests the asynchronous flow when it is ready

💯

**Pull request merged**